### PR TITLE
Add layers: DS-51

### PR DESCRIPTION
**Pull request by jonearley**

### CODE REVIEW

**Comment by jonearley** [comment]
Thoughts on this API?

`Z_DEPTH` or maybe it should be `SHADOWS`, `LAYERS`, `DEPTH`, ... something else? Docs for code are creeping into guides again...

Language around layers vs z-depth vs Z_DEPTH vs shadows seems a little all over the place. Maybe it could all be reduced to a single term?

**Comment by bridgetburke** [comment]
I think Z-DEPTH or SHADOWS. The first probably makes the most sense since the docs mention the z-axis and you do use y-spacing and other axis-specific styling.

File: packages/docs/src/pages/guides/layers/guidelines.mdx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+## Using shadows to create layers
+
+Elements with shadows can help users understand an interface. Shadows indicate hierarchy of elements by differentiating between two objects. In some cases, shadows help users to understand that one object is above another.
+
+## Guidelines
+- Use q **z-depth 1** as a hover effect for cards.
+- Use q **z-depth 2** for megamenus or other dropdown menus.
+- Allow surfaces to move in front of and behind other surfaces, such as scrolling the page behind a dropdown menu.
```
Comment (respinos) [inline packages/docs/src/pages/guides/layers/guidelines.mdx]: I'm curious about this --- the patterns I've seen with modals is to _prevent_ the background from scrolling behind the top of the stack.

**Comment by respinos** [comment]
`Z_DEPTH` works. I'd probably lean to `LAYERS` as an API, as it's less style-specific. `STACK` is probably too abstract? Do the z-depths have typical labels that can be used instead of numbers? (card, menu, modal?)

File: packages/docs/src/pages/guides/layers/guidelines.mdx (hunk 1 above)
Comment (jonearley) [reply packages/docs/src/pages/guides/layers/guidelines.mdx]: I wondered that too. I interpreted this as say you open the UniversalHeader dropdown menu then you should still be able to scroll on the page.

And agree that modals should prevent background scrolling.

**Comment by jonearley** [comment]
@bridgetburke @respinos Good points!

About using labels. An issue we could run into is adding layers between what we begin with if we incrementing by 1. So we have 1 and 2. What if we need something in between?

[Material UI uses `elevation`](https://material-components.github.io/material-components-web-catalog/#/component/elevation) with `dp` to select a depth where `dp` seems to correlate with a `box-shadow` value. That seems interesting. We should probably switch to a label that indicates usage like you're noting or use an equivalent to a `dp` that correlates to a box-shadow value so that if we ever add more we will have a "spot" for it.

**Comment by jonearley** [comment]
I prefer `LAYERS` over `Z_DEPTH` so far.

**Comment by jonearley** [comment]
localhost_8000_guides_layers_code

Updated `Z_DEPTH` to `Z_SPACE` since it's a distance between to layers its "space" and felt more like a _unit_ to use to access layers, to me at least.

Instead of `1` and `2`, updated to `8` and `16` to match the `box-shadow` `blur-radius` value. I think this should make this approach more flexible adding more depths, but these units don't indicate usage too well. I guess that comes through with the guidelines and less so in the labels.

File: packages/docs/src/pages/guides/layers/guidelines.mdx (hunk 1 above)
Comment (jonearley) [reply packages/docs/src/pages/guides/layers/guidelines.mdx]: Removed that confusing bullet for now.

**Comment by respinos** [comment]
I have no idea why `Z_SPACE` works for me, but it does! 
I suppose `Z.SPACE[Z.SERIOUS]` is awkward? (where `Z.SERIOUS` is a constant representing `8`)

**Pull request merged**